debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnection.java

```diff
@@ -336,9 +337,11 @@ else if (column.jdbcType() == OracleTypes.NUMBER) {
      * @throws IllegalStateException if the query does not return at least one row
      */
     public Scn getCurrentScn() throws SQLException {
-        return queryAndMap("SELECT CURRENT_SCN FROM V$DATABASE", (rs) -> {
+        return queryAndMap("SELECT CURRENT_SCN,scn_to_timestamp(CURRENT_SCN) FROM V$DATABASE", (rs) -> {
             if (rs.next()) {
-                return Scn.valueOf(rs.getString(1));
+                Scn scn = Scn.valueOf(rs.getString(1));
+                scn.setTime(rs.getObject(2, OffsetDateTime.class));
+                return scn;
```

Comment:
I think I would prefer we break out the `scn_to_timestamp` call into a separate method on the `OracleConnection` class similar to something like as follows:

```
public OffsetDateTime getScnToTimestamp(Scn scn) throws SQLException {
    return queryAndMap("SELECT scn_to_timestamp(" + scn.toString() + ") FROM DUAL", rs -> {
       if (rs.next()) {
         return rs.getObject(1, OffsetDateTime.class);
       }
       throw new DebeziumException("Failed to get SCN as a timestamp");
    });
}
```

So inside `LogMinerStreamingChangeEventSource` method we can use this special method to get the SCN as a time value so we can then do the comparison between the two SCN values easily to determine the time differential.